lib/ddtrace/contrib/mysql2/instrumentation.rb

```diff
@@ -27,36 +32,43 @@ def query(sql, options = {})
               span.set_tag(Datadog::Ext::Integration::TAG_PEER_SERVICE, span.service)
 
               # Set analytics sample rate
-              Contrib::Analytics.set_sample_rate(span, analytics_sample_rate) if analytics_enabled?
+              if Datadog.configuration[:mysql2][:analytics_enabled]
+                Contrib::Analytics.set_sample_rate(span, Datadog.configuration[:mysql2][:analytics_sample_rate])
+              end
 
               span.set_tag(Ext::TAG_DB_NAME, query_options[:database])
               span.set_tag(Datadog::Ext::NET::TARGET_HOST, query_options[:host])
               span.set_tag(Datadog::Ext::NET::TARGET_PORT, query_options[:port])
-              super(sql, options)
-            end
-          end
 
-          def datadog_pin
-            @datadog_pin ||= Datadog::Pin.new(
-              Datadog.configuration[:mysql2][:service_name],
-              app: Ext::APP,
-              app_type: Datadog::Ext::AppTypes::DB,
-              tracer: -> { Datadog.configuration[:mysql2][:tracer] }
-            )
+              # Invoke original method
+              yield(env)
+            end
           end
 
-          private
+          def get_client(env)
+            return unless env[:self].instance_of?(::Mysql2::Client)
```

Comment:
Should this be `is_a?`, rather than `instance_of?` to avoid coupling on the exact class, allowing some flexibility if we get a subclass? (see also https://stackoverflow.com/questions/3893278/ruby-kind-of-vs-instance-of-vs-is-a )